nucypher/blockchain/eth/agents.py

```diff
@@ -228,12 +230,34 @@ def partition_stakers_by_activity(self) -> Tuple[List[str], List[str], List[str]
 
         return active_stakers, pending_stakers, missing_stakers
 
-    def get_all_locked_tokens(self, periods: int) -> int:
-        """Returns the current period"""
+    def get_all_active_stakers(self, periods: int, pagination_size: int = None) -> Tuple[int, list]:
+        """Only stakers which confirmed the current period (in the previous period) are used."""
         if not periods > 0:
             raise ValueError("Period must be > 0")
 
-        all_locked_tokens, _stakers = self.contract.functions.getAllActiveStakers(periods).call()
+        if pagination_size is None:
+            pagination_size = StakingEscrowAgent.DEFAULT_PAGINATION_SIZE if self.blockchain.is_light else 0
+        elif pagination_size < 0:
+            raise ValueError("Pagination size must be >= 0")
+
+        if pagination_size > 0:
+            num_stakers = self.get_staker_population()
+            start_index = 0
+            n_tokens = 0
+            stakers = list()
+            while start_index < num_stakers:
+                temp_locked_tokens, temp_stakers = \
+                    self.contract.functions.getActiveStakers(periods, start_index, pagination_size).call()
+                n_tokens += temp_locked_tokens
+                stakers += temp_stakers
+                start_index += pagination_size
+        else:
+            n_tokens, stakers = self.contract.functions.getActiveStakers(periods, 0, 0).call()
```

Comment:
One interesting way could be to catch an error here and start paginating if the "heavy" call fails

Reply:
Interesting idea, I'd prefer more predictable solution and this method could fail with other errors (not in reality, just potentially), but auto pagination could work
if we don't need "light" flag then it definitely will be better

Reply:
I'd love to help on a dynamic pagination size. Sounds like a fun tiny project!

Reply:
It could be handy, but from development perspective I don't like it: for agent user better to understand what doing each method, but this method will be too smart. Also to do it correct we need to distinguish `TransactionFailed` from timeout problems